lib/putS3.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    logger,
+    Project,
+    ProjectFile,
+} from "@atomist/automation-client";
+import { doWithFiles } from "@atomist/automation-client/lib/project/util/projectUtils";
+import {
+    GoalInvocation,
+    ProgressLog,
+} from "@atomist/sdm";
+import { S3 } from "aws-sdk";
+import * as mime from "mime-types";
+import { PublishToS3Options } from "./options";
+
+type FilesAttempted = number;
+type SuccessfullyPushedKey = string;
```

Comment:
You define this two different places. Perhaps define once and import the other?

Reply:
the aliases are just for naming, not really doing anything. If they were types with some structure, I would.